test/test_linalg.py

```diff
@@ -2180,6 +2180,128 @@ def test_dot_invalid_args(self, device):
         self._test_dot_vdot_invalid_args(device, torch.dot)
         self._test_dot_vdot_invalid_args(device, torch.dot, complex_dtypes=True)
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_matrix_rank(self, device, dtype):
+        matrix_rank = torch.linalg.matrix_rank
+
+        def run_test(shape0, shape1, batch):
+            a = torch.randn(*batch, shape0, shape1, dtype=dtype, device=device)
+
+            self.assertEqual(matrix_rank(a), matrix_rank(a.conj().transpose(-2, -1)))
+            aaH = torch.matmul(a, a.conj().transpose(-2, -1))
+            self.assertEqual(matrix_rank(aaH), matrix_rank(aaH, hermitian=True))
```

Comment:
generally these tests are not guaranteed to pass due to tolerances, regular computation via svd and hermitian via symeig can sometimes produce different results, right?

Reply:
Probably svd and symeig could give slightly different results for the singular values, but I think the discrepancy should be smaller than the default tolerances used in `self.assertEqual`. It's usually the eigenvectors that could cause problems in such tests and we don't use them here.